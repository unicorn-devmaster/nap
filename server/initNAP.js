const {
  willAuthen,
  willLoginWithFacebook,
  willSignUp,
  willLogin,
  willLogout,
  willResetPassword
} = require('./authen')
const { willCreateUser } = require('./graphql/resolvers/UserResolver')
const { willInstall } = require('./graphql/resolvers/InstallationResolver')

const init = (req, res, next) => {
  // Inject req
  req.nap = req.nap || {
    willAuthen,
    willLoginWithFacebook,
    willSignUp,
    willLogin,
    willLogout,
    willResetPassword,
    willCreateUser,
    willInstall
  }

  // No errors
  req.nap.errors = []

  next()
}

module.exports = init
