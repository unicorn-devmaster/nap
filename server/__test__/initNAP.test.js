/* eslint-env jest */
describe('NAP Server', () => {
  test('initNAP', () => {
    const req = {}

    // Run
    require('../initNAP')(req, {}, () => { })
    expect(req).toMatchSnapshot()
  })
})
