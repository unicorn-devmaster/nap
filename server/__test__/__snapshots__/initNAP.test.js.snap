// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NAP Server initNAP 1`] = `
Object {
  "nap": Object {
    "errors": Array [],
    "willAuthen": [Function],
    "willCreateUser": [Function],
    "willInstall": [Function],
    "willLogin": [Function],
    "willLoginWithFacebook": [Function],
    "willLogout": [Function],
    "willResetPassword": [Function],
    "willSignUp": [Function],
  },
}
`;
